src/engine/blocks.js

```diff
@@ -463,18 +466,34 @@ class Blocks {
                 this.emitProjectChanged();
             }
             break;
+        case 'block_comment_collapse':
         case 'comment_collapse':
             if (this.runtime.getEditingTarget()) {
                 const currTarget = this.runtime.getEditingTarget();
                 if (currTarget && !Object.prototype.hasOwnProperty.call(currTarget.comments, e.commentId)) {
-                    log.warn(`Cannot change comment with id ${e.commentId} because it does not exist.`);
+                    log.warn(`Cannot collapse comment with id ${e.commentId} because it does not exist.`);
                     return;
                 }
                 const comment = currTarget.comments[e.commentId];
                 comment.minimized = e.newCollapsed;
                 this.emitProjectChanged();
             }
             break;
+        case 'block_comment_resize':
```

Comment:
So before the sizes of comments weren't saved?

Reply:
Previously that was handled in the comment_change event handler, which I removed [here](https://github.com/gonfunko/scratch-vm/commit/88c097b1f2b36c2cc244e977f8927c5d02783687#diff-15a5a47826a28e18ba1568342a197b4bc1b52dbbe3d7aefe3b503edc16ab9764L447-L458) because the comment change event just had a "newContents_" attribute, which was an object containing one of a string, size or a boolean to handle text, size, and collapse changes, and the modern workspace comment events (thankfully) don't work like that. I followed a similar pattern with the block comment events of distinct events per action, and re-added support for resizing both types of comments here.